RealmSwift/Tests/SectionedResultsTests.swift

```diff
@@ -0,0 +1,1911 @@
+        let section1 = sectionedResults[0]
+        let section2 = sectionedResults[1]
+
+        var firstRun = true
+        // Only get notifications for key 'a'.
+        let token1 = section1.observe(keyPaths: [\ModernAllTypesObject.stringCol]) { (changes: RealmSectionedResultsChange) in
+            switch changes {
+            case .initial(let collection):
+                XCTAssertEqual(collection.count, 1)
+            case let .update(collection, deletions: deletions, insertions: insertions, modifications: modifications,
+                             sectionsToInsert: sectionsToInsert, sectionsToDelete: sectionsToDelete):
+                if firstRun {
+                    XCTAssertEqual(sectionsToDelete, [0])
+                    XCTAssertTrue(sectionsToInsert.isEmpty)
+                    XCTAssertTrue(deletions.isEmpty)
+                    XCTAssertTrue(modifications.isEmpty)
+                    XCTAssertTrue(insertions.isEmpty)
+                } else {
+                    XCTAssertEqual(collection.count, 1)
+                    XCTAssertEqual(insertions, [IndexPath(item: 0, section: 0)])
+                    XCTAssertEqual(sectionsToInsert, [0])
+                }
+            case .error:
+                XCTFail("Shouldn't happen")
+            }
+        }
+
+        // Only get notifications for key 'b'.
+        let token2 = section2.observe(keyPaths: [\ModernAllTypesObject.stringCol]) { (changes: RealmSectionedResultsChange) in
+            switch changes {
+            case .initial(let collection):
+                if firstRun {
+                    XCTAssertEqual(collection.count, 2)
+                }
+            case let .update(collection, deletions: deletions, insertions: insertions, modifications: modifications,
+                             sectionsToInsert: sectionsToInsert, sectionsToDelete: sectionsToDelete):
+                if firstRun {
+                    XCTAssertEqual(collection.count, 1)
+                    XCTAssertTrue(sectionsToDelete.isEmpty)
+                    XCTAssertTrue(sectionsToInsert.isEmpty)
+                    XCTAssertEqual(deletions, [IndexPath(item: 0, section: 1)])
+                    XCTAssertTrue(modifications.isEmpty)
+                    XCTAssertTrue(insertions.isEmpty)
+                } else {
+                    XCTAssertEqual(collection.count, 2)
+                    XCTAssertTrue(sectionsToDelete.isEmpty)
+                    XCTAssertTrue(sectionsToInsert.isEmpty)
+                    XCTAssertEqual(deletions, [IndexPath(item: 0, section: 0)])
+                    XCTAssertTrue(modifications.isEmpty)
+                    XCTAssertEqual(insertions, [IndexPath(item: 0, section: 0), IndexPath(item: 1, section: 0)])
+                }
+            case .error:
+                XCTFail("Shouldn't happen")
+            }
+        }
+
+        try! realm.write {
+            realm.delete([results[2], results[0]]) // banana, apple
```

Comment:
After this delete is made, it looks like the "a" section has the delete notify as a `sectionsToDelete` (490), because apple was the last "a". But the "b" section notification propagates as `deletions`(518) with an index, because "box" remains in the section. Is that right? 
Is there a way for the developer to see what the last object(s) of a section were in a deletion?

Reply:
'Is that right?'– Yes that is correct.
'Is there a way for the developer to see what the last object(s) of a section were in a deletion?'- This would require setting up a realm with a transaction ref pointing to the previous state. We don't expose anything like that. Maybe if they really needed that they could hold a frozen instance of the sectioned results before their modifications and use that later on.